passage_error.go

```diff
@@ -1,29 +1,48 @@
 package passage
 
 import (
+	"encoding/json"
 	"fmt"
 	"strings"
 )
 
 type PassageError struct {
 	Message    string
-	StatusCode int
 	ErrorCode  string
+	StatusCode int
 }
 
 func (e PassageError) Error() string {
-	var ps strings.Builder
-	ps.WriteString("Passage Error - ")
+	var sb strings.Builder
+	sb.WriteString("PassageError - ")
 
 	if e.Message != "" {
-		fmt.Fprintf(&ps, "message: %s, ", e.Message)
+		sb.WriteString(fmt.Sprintf("message: %s, ", e.Message))
+	}
+
+	if e.ErrorCode != "" {
+		sb.WriteString(fmt.Sprintf("errorCode: %v, ", e.ErrorCode))
 	}
+
 	if e.StatusCode != 0 {
-		fmt.Fprintf(&ps, "status_code: %v, ", e.StatusCode)
+		sb.WriteString(fmt.Sprintf("statusCode: %v, ", e.StatusCode))
 	}
-	if e.ErrorCode != "" {
-		fmt.Fprintf(&ps, "error_code: %s, ", e.ErrorCode)
+
+	return strings.TrimSuffix(sb.String(), ", ")
+}
+
+func errorFromResponse(body []byte, statusCode int) error {
```

Comment:
since each generated client call returns its own struct (i.e. `GetUserResponse`, `ActivateUserResponse`, etc) and the required values are only exposed as properties (`JSON400.Error` or `JSON400.Code`)

it can't be easily abstracted into a generic function without delineating each of the different possible types and casting them.

so instead this function manually unmarshalls the response body instead of using the pre-parsed `JSONXXX` properties to DRY up the code.

the trade-off is that it's taking extra steps to parse the bytes into a new struct, but i thought the simplification was worth it.